ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -764,23 +777,26 @@ module Nvidia_compat = struct
     read_configs []
       (List.map (fun conf -> Filename.concat conf_dir conf) conf_files)
 
-  let create_compat_lookup_file __context =
+  let create_compat_config_file __context =
     try
       let open Db_filter_types in
-      read_config_dir conf_dir
-      |> List.map (fun (conf_file, identifier) ->
-        let identifier_string = Identifier.to_string identifier in
-        let expr = Eq (Field "identifier", Literal identifier_string) in
-        match Db.VGPU_type.get_internal_records_where ~__context ~expr with
-        | [_, rc] ->
-          let type_id = List.assoc Xapi_globs.vgpu_type_id rc.Db_actions.vGPU_type_internal_config in
-          Printf.sprintf "%s %s\n" conf_file type_id
-        | _ -> "" (* Type is not relevant: ignore *)
-      )
-      |> String.concat ""
-      |> Stdext.Unixext.write_string_to_file !Xapi_globs.nvidia_compat_lookup_file
+      let host_driver_version = Vendor_nvidia.get_host_driver_version () in
+      let host_driver_support_multiple = Vendor_nvidia.is_host_driver_support_multi_vgpu ~host_driver_version ~supported_versions:!Xapi_globs.nvidia_multi_vgpu_enabled_driver_versions in
+      if not host_driver_support_multiple then
+        read_config_dir conf_dir (* Nvidia host driver does not support multiple vGPU, create compat config file *)
+        |> List.iter (fun (conf_file, identifier) ->
+            let identifier_string = Identifier.to_string identifier in
```

Comment:
Maybe just `id`

Reply:
all other parts use `identifier_string` instead of `id`,  like 
```145 let find_and_update ~__context vgpu_type =
  146   let identifier_string = Identifier.to_string vgpu_type.identifier in
```
maybe we shall keep consistency.